### PR TITLE
fix(provisioning): export useful namesByAddress (release branch)

### LINKED ISSUE #8408: E(namesByAddressAdmin).readonly() throws pattern error: promise vs. remotable
## Describe the bug

I was running the agoric-upgrade-11wf and faced the patternMatcher problem of the E(namesByAddressAdmin).readonly() again. 

Here's the stack trace:
```
2023-09-27T19:15:37.565Z SwingSet: ls: v11: Logging sent error stack (Error#1)
2023-09-27T19:15:37.565Z SwingSet: ls: v11: Error#1: In "readonly" method of (NamesByAddressAdmin): result: promise (an object) - Must be a remotable
2023-09-27T19:15:37.565Z SwingSet: ls: v11: Error: In "readonly" method of (NamesByAddressAdmin): result: promise (an object) - Must be a remotable
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at throwLabeled (/bundled-source/.../node_modules/@endo/patterns/src/utils.js:132)
 at applyLabelingError (/bundled-source/.../node_modules/@endo/patterns/src/utils.js:154)
 at mustMatch (/bundled-source/.../node_modules/@endo/patterns/src/patterns/patternMatchers.js:528)
 at In "readonly" method of (NamesByAddressAdmin) (/bundled-source/.../node_modules/@endo/exo/src/exo-tools.js:43)
 at apply ()
 at apply ()
 at dispatchToHandler (/bundled-source/.../node_modules/@endo/eventual-send/src/handled-promise.js:163)
 at win (/bundled-source/.../node_modules/@endo/eventual-send/src/handled-promise.js:511)
 at ()
```

We need to send some invitations the core-eval ; And we can't get the depositFacets for those accounts

e.g.
const namesByAddress = E(namesByAddressAdmin).readonly()
patternMatcher throws error. 
It says the method 'readonly' should return a remotable but it returns a promise

Developer used the following fix to get it working in their local environment but it did not work in DevNet
https://github.com/Agoric/agoric-sdk/blob/31ed40207648bb76e2b37cb82dd068fb1837fdd8/packages/vats/src/nameHub.js#L44
(change it to readonly: M.call().returns(M.any())

## Platform Environment
 - what OS are you using? what version of Node.js?
 - is there anything special/unusual about your platform?
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)

## Additional context
Read the full thread discussion
https://discord.com/channels/585576150827532298/755164472397660313/1156672044608532590

## Screenshots
If applicable, add screenshots to help explain your problem, especially for UI interactions.

### PR DESCRIPTION
**Pull request by dckc**
closes: #8311
this fix is for the release branch, anyway.
see #8821 for fix on master

DRAFT until:
 - [x] e2e test that namesByAddress works for delivering stuff
 - [x] figure out how to get CI to run
 - [x] review bootstrap for consumers of `namesByAddress`
 - [ ] figure out end-to-end testing infrastructure and release branch stuff
   - stacked on: #8819
   - see also: https://github.com/Agoric/agoric-3-proposals/pull/76

## Description

The `namesByAddress` provided by the provisioning vat is disconnected from `namesByAddressAdmin`, so it never gets populated.

Fortunately, the relevant `nameHubKit` is in baggage. So the fix here is to return that one.

### Security Considerations

This should make it unnecessary for future core-eval scripts to get the whole `namesByAddressAdmin` authority when they only need `namesByAddress` to do something like deliver fees.

The permit of the upgrade core-eval is:

```js
      consume: {
        vatStore: true,
        vatAdminSvc: true,
      },
      produce: {
        namesByAddress: true,
      },
```

**Read/write access to the whole `vatStore` is excess authority; we only need access to read one of its entries, not authority to scribble over all the others and upgrade all the other vats.**
`vatAdminSvc` is needed to look up a bundleID (hash) and return the corresponding bundleCap. Its only method beyond that sort of bundle lookup is `E(vatAdminSvc).createVat()`. That method is more than we need, but while it can be wasteful, it doesn't pose any risk to integrity.

### Scaling Considerations

n/a

### Documentation Considerations

fixes the chain to agree with our [docs on name services](https://docs.agoric.com/guides/integration/name-services.html)

### Testing Considerations

see DRAFT until list above

### Upgrade Considerations

The bug is localized to the provisioning vat, so upgrading it seems to suffice.

Perhaps we should fix #8408 while we're at it?

### CODE REVIEW

**Comment by dckc** [comment]
seems to work...

I wonder what all the **failed deposit to stATOM-USD price feed member** are about. I hope it's left over from experimenting with other things on this local chain.

```
ag-power-tools-agd-1  | 2024-01-22T18:30:57.130Z SwingSet: vat: v1: evaluateBundleCap { manifestBundleRef: { bundleID: 'b1-af71cf8c6c422525b406913a427b6d6afe93bd29929f64a5e115da0913f37342931ac58abf3a60f5144ae642e285544abec2e637312d3f8802f13e401346af5b' }, exportedGetManifest: 'getManifestForProvisioning', vatAdminSvc: Promise [Promise] {} }
ag-power-tools-agd-1  | 2024-01-22T18:30:57.182Z SwingSet: vat: v1: execute { exportedGetManifest: 'getManifestForProvisioning', behaviors: [ 'getManifestForProvisioning', 'upgradeProvisioningVat' ] }
ag-power-tools-agd-1  | 2024-01-22T18:30:57.183Z SwingSet: vat: v1: coreProposal: upgradeProvisioningVat
ag-power-tools-agd-1  | 2024-01-22T18:30:57.183Z SwingSet: vat: v1: upgradeProvisioningVat...
ag-power-tools-agd-1  | 2024-01-22T18:30:57.184Z SwingSet: vat: v1: provisioning { adminNode: Object [Alleged: adminNode] {}, root: Object [Alleged: undefined] {} } b1-c40cda8e60deec8147a547ccfda551398056d69280edbf345e25f75c7943cd3b597637b9761ceb9d3e4690721be69dd4f493bab39177dbd98ee9633a41549f8d
ag-power-tools-agd-1  | 2024-01-22T18:30:58.039Z SwingSet: kernel: vat v11 upgraded from incarnation 0 to 1 with source b1-c40cda8e60deec8147a547ccfda551398056d69280edbf345e25f75c7943cd3b597637b9761ceb9d3e4690721be69dd4f493bab39177dbd98ee9633a41549f8d
ag-power-tools-agd-1  | 2024-01-22T18:30:58.043Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric1ldmtatp24qlllgxmrsjzcpe20fvlkp448zcuce { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.044Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric140dmkrz2e42ergjj7gyvejhzmjzurvqeq82ang { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.046Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric1w8wktaur4zf8qmmtn3n7x3r0jhsjkjntcm3u6h { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.047Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric10vjkvkmpp9e356xeh6qqlhrny2htyzp8hf88fk { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.049Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric1qj07c7vfk3knqdral0sej7fa6eavkdn8vd8etf { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.050Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric1lw4e4aas9q84tq0q92j85rwjjjapf8dmnllnft { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.051Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric1ra0g6crtsy6r3qnpu7ruvm7qd4wjnznyzg5nu4 { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.052Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric1zj6vrrrjq4gsyr9lw7dplv4vyejg3p8j2urm82 { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.053Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric15xddzse9lq74cyt6ev9d7wywxerenxdgxsdc3m { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.054Z SwingSet: vat: v1: failed deposit to stATOM-USD price feed member agoric1w5wmck6q2xrt20ax3njlk2k87m4t4l2y2xgw2d { incarnationNumber: 0, name: 'vatUpgraded', upgradeMessage: 'fix namesByAddress' }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.058Z SwingSet: vat: v1: { name: 'provisioning', opts: { upgradeMessage: 'fix namesByAddress' }, incarnation: { incarnationNumber: 1 } }
ag-power-tools-agd-1  | 2024-01-22T18:30:58.236Z SwingSet: vat: v1: namesByAddress: new Promise
ag-power-tools-agd-1  | 2024-01-22T18:30:58.237Z SwingSet: vat: v1: namesByAddress settled; remaining: [ 'anchorKits', 'bankMints', 'contractGovernor', 'reserveGovernorCreatorFacet', 'scaledPriceAuthorityKits', 'vBankKits', 'vaultFactoryGovernorCreator' ]
```

**Comment by dckc** [comment]
> -  review bootstrap for consumers of namesByAddress

I reviewed all occurrences of `namesByAddress` in the current SDK; unless there's something sneaky and dynamic going on, no production contracts nor vats are holding on to the old one.

**Comment by dckc** [comment]
@turadg please take a look. It's **draft** only because it's stacked on #8819 , my work-in-progress port of a3p-integration with "simplify core-eval" to a release branch context.

@mhofman perhaps you've got thoughts on how this should land in the release branch.

cc @Chris-Hibbert who is doing similar stuff.

See also #8821 for the corresponding fix for master, especially the testing considerations.

File: a3p-integration/proposals/c:names-by-addr/post.test.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    .map(([k, v]) => [`--${k}`, v])
+    .flat();
+};
+
+export const txAbbr = tx => {
+  const { txhash, code, height, gas_used } = tx;
+  return { txhash, code, height, gas_used };
+};
+
+// XXX overlaps with passCoreEvalProposal from synthetic chain
```
Comment (turadg) [inline a3p-integration/proposals/c:names-by-addr/post.test.js:40]: why not use it?
https://github.com/Agoric/agoric-3-proposals/blob/6e5fec521b33a3479ba9a6fd198b9313f49c5a7f/packages/synthetic-chain/src/lib/core-eval.ts#L86-L89

File: a3p-integration/proposals/c:names-by-addr/test.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+source /usr/src/upgrade-test-scripts/env_setup.sh
+
+yarn ava
```
Comment (turadg) [inline a3p-integration/proposals/c:names-by-addr/test.sh:4]: glad see no file specified

File: packages/vats/test/bootstrapTests/test-demo-config.js [hunk 3]
```diff
@@ -67,6 +67,16 @@ test('sim/demo config provides home with .myAddressNameAdmin', async t => {
   keyArrayEqual(t, keys(home).sort(), homeKeys);
 });
 
+test('namesByAddress contains provisioned account', async t => {
```
Comment (turadg) [inline packages/vats/test/bootstrapTests/test-demo-config.js:70]: good thinking including here too

File: a3p-integration/proposals/c:names-by-addr/post.test.js (hunk 1 above)
Comment (dckc) [reply a3p-integration/proposals/c:names-by-addr/post.test.js:40]: hm... the reason has faded. something about staticConfig? I'll have to look again.

**Comment by dckc** [comment]
overtaken by stuff such as ##8901

**Pull request closed without merging**